File: src.csharp/AlphaTab/Core/Dom/TextEncoder.cs
using System.Text;
using AlphaTab.Core.EcmaScript;

namespace AlphaTab.Core.Dom
{
    internal class TextEncoder
    {
        private readonly Encoding _encoding;

        public TextEncoder()
        {
            _encoding = Encoding.UTF8;
        }

        public Uint8Array Encode(string value)
        {
            return _encoding.GetBytes(value);
        }
    }
}

File: src.kotlin/alphaTab/shared/src/commonMain/kotlin/alphaTab/core/dom/TextEncoder.kt
package alphaTab.core.dom

internal class TextEncoder {
    @ExperimentalUnsignedTypes
    public fun encode(str: String): alphaTab.core.ecmaScript.Uint8Array {
        return alphaTab.core.ecmaScript.Uint8Array(str.encodeToByteArray().toUByteArray())
    }
}
